### PR TITLE
SDK-468 [Python SDK] add caching support for /countries data

### PR DESCRIPTION
**Pull request by RommeroInCountry**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-468](https://incountry.atlassian.net/browse/SDK-468)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: incountry/countries_cache.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+from .exceptions import StorageCountryNotSupportedException
+from .models import HttpOptions
+from .validation import validate_http_response
+from .http_request import http_request
+
+
+class CountriesCache:
+    DEFAULT_COUNTRIES_ENDPOINT = "https://portal-backend.incountry.com/countries"
+    DEFAULT_CACHE_TTL = 60
```
Comment (VladimirLevitskiy) [inline incountry/countries_cache.py:13]: 300 seconds are used in other SDKs

File: incountry/countries_cache.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            "headers": request_options.get("http_headers", {}),
+            "timeout": self.options.timeout,
+        }
+        return http_request(params, scope="fetching countries list")
+
+    def get_country_details(self, country):
+        country = country.lower()
+
+        if self.cache_expires_at <= time.time():
+            self.update_countries_cache()
```
Comment (VladimirLevitskiy) [inline incountry/countries_cache.py:57]: Make it failsafe. If PB returns an empty country list response, use previous countries cache

**Review by VladimirLevitskiy — CHANGES_REQUESTED** [review]

File: incountry/countries_cache.py (hunk 2 above)
Comment (RommeroInCountry) [reply incountry/countries_cache.py:57]: @VladimirLevitskiy completely ignore if /countries response validation fails?

**Review by VladimirLevitskiy — DISMISSED** [review]

**Review by VladimirLevitskiy — APPROVED** [review]

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Pull request merged**